src/mux.rs

```diff
@@ -191,12 +191,14 @@ impl<T: TokioConn> Future for MuxTimer<T> {
             let ts = get_timestamp_slow();
             self.timestamp.store(ts, Ordering::SeqCst);
             let mut state = self.state.lock();
+            let mut last_ping = self.last_ping.lock();
 
             // Ping
             if let Some(keep_alive_interval) = self.keep_alive_interval {
-                if ts > self.last_ping + keep_alive_interval {
+                if ts > *last_ping + keep_alive_interval {
```

Comment:
Please, lock ping mutex only here - no need to lock it for mode without timeout. And for a long range of code